http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala

```diff
@@ -494,17 +494,41 @@ class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with Flat
           gdDAO.clusters should contain key clusterName
         }
 
-        // change cluster status to Running so that it can be deleted
-        dbFutureValue { _.clusterQuery.setToRunning(cluster.id, IP("numbers.and.dots")) }
+        // change cluster status to Error
+        dbFutureValue { _.clusterQuery.updateClusterStatus(cluster.id, ClusterStatus.Error) }
 
         // delete the cluster
         leo.deleteCluster(userInfo, project, clusterName).futureValue
 
         // check that the cluster no longer exists
         gdDAO.clusters should not contain key(clusterName)
+
+        // cluster should be in Deleting state (the cluster monitor transitions it to Deleted)
+        dbFutureValue { _.clusterQuery.getClusterStatus(cluster.id) } shouldBe Some(ClusterStatus.Deleting)
     }
   }
 
+  it should "delete a cluster that has status Error and does not have a google id" in isolatedDbTest {
```

Comment:
Where are we setting up the cluster to not have a google id?

Reply:
Right. I first thought a cluster not having a Google ID was a condition to reproduce the bug this PR fixes, in which case, we'd wanna incorporate that into the test as set-up but it's actually sufficient to have a cluster in Error state to run into the bug while trying to delete it?

Reply:
Yeah good idea.. will add.

Reply:
Added a verification for this.